src/EFCore.PG/Query/Expressions/Internal/PgFunctionExpression.cs

```diff
@@ -252,69 +273,84 @@ protected override Expression VisitChildren(ExpressionVisitor visitor)
                 : this;
         }
 
-        /// <summary>
-        ///     Tests if this object is considered equal to another.
-        /// </summary>
-        /// <param name="obj"> The object to compare with the current object. </param>
-        /// <returns>
-        ///     true if the objects are considered equal, false if they are not.
-        /// </returns>
-        public override bool Equals(object obj)
+        /// <inheritdoc />
+        public override bool Equals(object obj) => obj is PgFunctionExpression pgFunction && Equals(pgFunction);
+
+        /// <inheritdoc />
+        public bool Equals(PgFunctionExpression other)
+            => other != null &&
+               Instance == other.Instance &&
+               Type == other.Type &&
+               FunctionName == other.FunctionName &&
+               Schema == other.Schema &&
+               _positionalArguments.Count == other._positionalArguments.Count &&
```

Comment:
I'm pretty sure `SequenceEquals` below already verifies that the counts are identical

Reply:
As above, will revert.